internal/fleet/integration_policy/models.go

```diff
@@ -91,12 +92,45 @@ func (model *integrationPolicyModel) populateFromAPI(ctx context.Context, data *
 	model.IntegrationVersion = types.StringValue(data.Package.Version)
 	model.VarsJson = utils.MapToNormalizedType(utils.Deref(data.Vars), path.Root("vars_json"), &diags)
 
+	// Preserve space_ids if it was originally set in the plan/state
+	// The API response may not include space_ids, so we keep the original value
+	originallySetSpaceIds := utils.IsKnown(model.SpaceIds)
+	if data.SpaceIds != nil {
+		spaceIds, d := types.ListValueFrom(ctx, types.StringType, *data.SpaceIds)
+		diags.Append(d...)
+		model.SpaceIds = spaceIds
+	} else if !originallySetSpaceIds {
+		// Only set to null if it wasn't originally set
+		model.SpaceIds = types.ListNull(types.StringType)
+	}
+	// If originally set but API didn't return it, keep the original value
+
 	model.populateInputFromAPI(ctx, data.Inputs, &diags)
 
 	return diags
 }
 
 func (model *integrationPolicyModel) populateInputFromAPI(ctx context.Context, inputs map[string]kbapi.PackagePolicyInput, diags *diag.Diagnostics) {
+	// Handle input population based on context:
+	// 1. If model.Input is unknown: we're importing or reading fresh state → populate from API
+	// 2. If model.Input is known and null/empty: user explicitly didn't configure inputs → don't populate (avoid inconsistent state)
+	// 3. If model.Input is known and has values: user configured inputs → populate from API
+
+	isInputKnown := utils.IsKnown(model.Input)
+	isInputNullOrEmpty := model.Input.IsNull() || (isInputKnown && len(model.Input.Elements()) == 0)
+
+	// Case 1: Unknown (import/fresh read) - always populate
+	if !isInputKnown {
+		// Import or fresh read - populate everything from API
+		// (continue to normal population below)
+	} else if isInputNullOrEmpty {
```

Comment:
```suggestion
	if model.Input.IsNull() {
```

Couple of questions:
* Can we just do away with the first condition here? It's not clear what it's preventing from falling through to the empty case
* `ifInputKnown && len() == 0`. `input` is currently defined as a block, so I'm not sure what the behaviour is here. I'd expect `isInputKnown` means that the user has defined an `input` somewhere, and that if that collection is empty then it's been explicitly defined as empty and we should populate the model. I assume that's wrong though since you've written this code, is there an example of a TF config which hits each of these cases?